public/scripts/dom-utils.js

```diff
@@ -19,6 +19,16 @@ export function toggleVisibleElement($el) {
   }
 }
 
+export function sortChildren($el, attribute) {
+  [...$el.children]
+    .sort(
+      (a, b) =>
+        parseInt(a.getAttribute(attribute)) -
+        parseInt(b.getAttribute(attribute)),
+    )
+    .forEach((node) => $el.appendChild(node));
+}
```

Comment:
Here is a improved async version of this function to do the ordering instantly:

```suggestion
export async function sortChildren($el, attribute) {
  const sorted = [...$el.children]
    .sort(
      (a, b) =>
      parseInt(a.getAttribute(attribute)) -
      parseInt(b.getAttribute(attribute)),
    )
  $el.innerHTML = ""
  sorted.slice(0, 15).forEach((node) => $el.appendChild(node))
  setTimeout(async () => {
    sorted.slice(15).forEach((node) => $el.appendChild(node));
  }, 5);
}
```

Resetting the HTML content and appending synchronously only the first 15 icons do the trick. I suppose that this number can be increased without decreasing performance.

Reply:
Isn't setTimeout going to create a new anon function per 15 elements, thus eating memory?
What happens if you sort after having scrolled down a bit (thus the elements you see are not the first 15)?

Wasn't it simpler to put an overlay while sorting instead of doing this?

Reply:
> Isn't setTimeout going to create a new anon function per 15 elements, thus eating memory?

Not, `slice(15)` means from 15 to the end. `setTimeout` is executed one time.

> What happens if you sort after having scrolled down a bit (thus the elements you see are not the first 15)?

That is the problem. The page scrolls to the top when the ordering is changed. Not sure if could be worth to store the scroll height pixel and reset the same scroll position after ordering. Also can be added a `overflow: hidden` to the body during the change to hide the scroll bar meanwhile to not show how changes its size.